src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -74,16 +71,17 @@ class IqPolicyEvaluatorUtil
 
       return evaluationResult
     }
-    catch (IqNetworkException e) {
+    catch (IqClientException e) {
       return handleNetworkException(iqPolicyEvaluator.failBuildOnNetworkError, e, listener, run)
     }
   }
 
-  private static handleNetworkException(final boolean failBuildOnNetworkError, final IqNetworkException e,
-                                    final TaskListener listener, final Run run)
+  private static handleNetworkException(final Boolean failBuildOnNetworkError, final IqClientException e,
+                                        final TaskListener listener, final Run run)
   {
-    if (failBuildOnNetworkError) {
-      throw e.cause
+    def isNetworkError = isNetworkError(e)
+    if (!isNetworkError || isNetworkError && failBuildOnNetworkError) {
```

Comment:
Isn't this the same as `!isNetworkError || failBuildOnNetworkError`?

Reply:
d'oh 84cf567